### PR TITLE
Correctly mark unannotated NamedTuple field to be inferred TensorType

### LINKED ISSUE #46326: [JIT] Unannotated NamedTuple Type should use TensorType::getInferred()
## 🐛 Bug / 🚀  Feature
```
import torch
from collections import namedtuple
_MyNamedTuple = namedtuple('_MyNamedTuple', ['value'])

@torch.jit.script
def foo():
    return _MyNamedTuple(1)
> Expected a value of type 'Tensor' for argument 'value' but instead found type 'int'.
```

It would be nice if we showed that the Tensor type was inferred because no type annotations were given for the named tuple. There was a user internally who was confused by this.

cc @gmagogsfm

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Summary: If there is no annotation given, we want to show users that the type is inferred

Test Plan: Added a new test case that throws an error with the expected error message

Reviewers: Yanan Cao

Subscribers:

Tasks: 

Tags:

Fixes #46326

### CODE REVIEW

File: torch/_jit_internal.py [hunk 1]
```diff
@@ -964,14 +964,14 @@ def container_checker(obj, target_type):
 
 
 def _isinstance(obj, target_type) -> bool:
-    origin_type = get_origin(target_type)
```
Comment (gmagogsfm) [inline torch/_jit_internal.py:967]: Maybe make these cosmetic changes in a separate PR?

File: test/test_jit.py [hunk 2]
```diff
@@ -13796,6 +13796,11 @@ def test_non_primitive_types(x):
         out = test_non_primitive_types(_MyNamedTuple(value=torch.tensor(5.0)))
         self.assertEqual(out, torch.tensor(6.0))
 
+        with self.assertRaisesRegex(RuntimeError, 'inferred'):
```
Comment (gmagogsfm) [inline test/test_jit.py:13799]: Though we are indeed trying to check that tensor type has `inferred` set to true, it is better if we check more than just `inferred` in the error message.

Also, it is generally better to minimize amount of code under `self.assertRaisesRegex`, so I would suggest define `test_error` outside and only call `torch.jit.script` on it inside `self.assertRaisesRegex` scope.

**Review by gmagogsfm — COMMENTED** [review]
Nice work, just a few minor comments

File: test/test_jit.py (hunk 2 above)
Comment (tugsbayasgalan) [reply test/test_jit.py:13799]: Got it!

File: torch/_jit_internal.py (hunk 1 above)
Comment (tugsbayasgalan) [reply torch/_jit_internal.py:967]: No idea why this change happened. I will remove them for now.

File: test/test_jit.py [hunk 3]
```diff
@@ -13796,6 +13796,15 @@ def test_non_primitive_types(x):
         out = test_non_primitive_types(_MyNamedTuple(value=torch.tensor(5.0)))
         self.assertEqual(out, torch.tensor(6.0))
 
+        def test_error():
```
Comment (gmagogsfm) [inline test/test_jit.py:13799]: I think this deserves to be its own test case rather than being part of `test_isinstance_metacompile`

Comment (tugsbayasgalan) [reply test/test_jit.py:13799]: got it

**Review by gmagogsfm — APPROVED** [review]
